cf-custom-resources/lib/trigger-state-machine.js

```diff
@@ -0,0 +1,135 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+"use strict";
+
+const aws = require("aws-sdk");
+
+/**
+ * Main handler, invoked by Lambda
+ */
+exports.handler = async function (event, context) {
+  const responseData = {};
+  const physicalResourceId = event.PhysicalResourceId || event.LogicalResourceId;
+
+  const handler = async function () {
+    switch (event.RequestType) {
+      case "Create":
+      case "Update":
+        const sf = new aws.StepFunctions();
+        const params = {
+          stateMachineArn: event.ResourceProperties.StateMachineARN,
+          input: JSON.stringify({
+            SourceBucket: event.ResourceProperties.SourceBucket,
+            AssetMappingFilePath: event.ResourceProperties.AssetMappingFilePath,
+            DestinationBucket: event.ResourceProperties.DestinationBucket,
+          })
+        };
+
+        const res = await sf.startSyncExecution(params).promise();
+        if (res.status === "FAILED") {
+          throw new Error(`State machine failed: ${res.cause}`);
```

Comment:
Do you have an example of what this looks like?

My main worry is that if the `copy-asset` fn fails for one of the objects, how easy will it be to troubleshoot for the customer what exactly happened? will they have to go to the SFN console or can we display enough info from the progress tracker?

Reply:
Ok these are the three error classes I can think of, and a screenshot of what the error looks like:
1. Trigger state machine lambda fails (no permission to execute state machine)
<img width="920" alt="Screen Shot 2023-05-12 at 13 08 50" src="https://github.com/aws/copilot-cli/assets/10566468/b7e21835-6577-49dc-99eb-849db5e69a98">

2. State machine fails (no permission to get asset mapping file)
<img width="936" alt="Screen Shot 2023-05-12 at 13 18 09" src="https://github.com/aws/copilot-cli/assets/10566468/8ec57abc-539d-42f2-a054-dd5ad0072af7">

3. Copy lambda fails (no permission to Put in dest bucket)
<img width="932" alt="Screen Shot 2023-05-12 at 13 32 37" src="https://github.com/aws/copilot-cli/assets/10566468/2c7730d5-185a-4468-9d5a-19cebf5eb2f7">